Tests/Integration/Bootstrap/UpdateTest.php

```diff
@@ -0,0 +1,101 @@
+        $this->manager->getConnection()->executeQuery('DELETE FROM s_plugin_connect_product_to_categories');
+        $this->manager->getConnection()->executeQuery('DELETE FROM s_plugin_connect_items');
+
+        $categories = [
+            '/deutsch' => 'Deutsch',
+            '/deutsch/test1' => 'Test 1',
+            '/deutsch/test1/test11' => 'Test 1.1',
+            '/deutsch/test2' => 'Test 2',
+        ];
+        $categoriesJSON = json_encode($categories);
+
+        $this->manager->getConnection()->executeQuery('INSERT INTO s_plugin_connect_items (article_id, exported, category) VALUES (?, ?, ?)',
+            [3, null, $categoriesJSON]
+        );
+
+        $this->invokeMethod($this->update, 'recreateRemoteCategoriesAndProductAssignments');
+
+        $result = $this->manager->getConnection()->executeQuery('SELECT COUNT(*) AS number FROM s_plugin_connect_categories')->fetch();
+        $this->assertEquals('4', $result['number']);
+
+        $result = $this->manager->getConnection()->executeQuery('SELECT COUNT(*) AS number FROM s_plugin_connect_product_to_categories WHERE articleID = 3')->fetch();
+        $this->assertEquals('4', $result['number']);
+    }
+
+    public function testRecreateRemoteCategoriesAndProductAssignmentsAssignesAllCategories()
+    {
+        $this->manager->getConnection()->executeQuery('DELETE FROM s_plugin_connect_categories');
+        $this->manager->getConnection()->executeQuery('DELETE FROM s_plugin_connect_product_to_categories');
+        $this->manager->getConnection()->executeQuery('DELETE FROM s_plugin_connect_items');
+
+        $categories = [
+            '/deutsch' => 'Deutsch',
+            '/deutsch/test1' => 'Test 1',
+            '/deutsch/test1/test11' => 'Test 1.1',
+            '/deutsch/test2' => 'Test 2',
+        ];
+
+        foreach ($categories as $categoryKey => $category) {
+            $this->manager->getConnection()->executeQuery('INSERT INTO s_plugin_connect_categories (category_key, label) VALUES (?, ?)',
+                [$categoryKey, $category]);
+        }
+
+        $categoriesJSON = json_encode($categories);
+
+        $this->manager->getConnection()->executeQuery('INSERT INTO s_plugin_connect_items (article_id, exported, category) VALUES (?, ?, ?)',
+            [3, null, $categoriesJSON]
+        );
+
+        $this->invokeMethod($this->update, 'recreateRemoteCategoriesAndProductAssignments');
+
+        $result = $this->manager->getConnection()->executeQuery('SELECT COUNT(*) AS number FROM s_plugin_connect_categories')->fetch();
+        $this->assertEquals('4', $result['number']);
+
+        $result = $this->manager->getConnection()->executeQuery('SELECT COUNT(*) AS number FROM s_plugin_connect_product_to_categories WHERE articleID = 3')->fetch();
+        $this->assertEquals('4', $result['number']);
+    }
+
+    public function invokeMethod(&$object, $methodName, array $parameters = [])
```

Comment:
Did you try to run $this->update->run? You're able to modify SwagConnect version in DB then only needed private methods will be executed. In your case just update version to 1.1.4 in Db.

Reply:
the version is passed in the constructor to the updater so no need to change something in the DB.
Works like a charm :+1: